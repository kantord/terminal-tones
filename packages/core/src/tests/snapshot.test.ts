import path from "path";
import { it, expect, describe } from 'vitest';
import { generateColorScheme } from "..";

function isValidHexColor(str) {
  return /^#(?:[0-9a-fA-F]{3}){1,2}$/.test(str);
}

describe('generateColorScheme()', () => {
  it('returns the correct values', async () => {
    const imagePath = path.join(__dirname, "images", "image1.jpg")

    const results = await generateColorScheme(imagePath)

    expect(results).toMatchSnapshot()
    expect(results.terminal).toHaveLength(16)

    for (const color of results.terminal) {
      expect(isValidHexColor(color)).toBe(true)
    }

  })
})
