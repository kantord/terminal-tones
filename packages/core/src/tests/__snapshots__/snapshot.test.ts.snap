// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateColorScheme() > applies lightness multiplier to background 1`] = `
{
  "a": "#11160e",
  "b": "#1d231a",
}
`;

exports[`generateColorScheme() > returns the correct values 1`] = `
{
  "contrastColors": [
    {
      "background": "#11160e",
    },
    {
      "name": "neutral",
      "values": [
        {
          "contrast": 1,
          "name": "neutral100",
          "value": "#11180e",
        },
        {
          "contrast": 2,
          "name": "neutral200",
          "value": "#3f4b42",
        },
        {
          "contrast": 3,
          "name": "neutral300",
          "value": "#57655e",
        },
        {
          "contrast": 4,
          "name": "neutral400",
          "value": "#70786b",
        },
        {
          "contrast": 5,
          "name": "neutral500",
          "value": "#81877d",
        },
        {
          "contrast": 6,
          "name": "neutral600",
          "value": "#8f958c",
        },
        {
          "contrast": 7,
          "name": "neutral700",
          "value": "#9ca199",
        },
        {
          "contrast": 8,
          "name": "neutral800",
          "value": "#a8aca6",
        },
        {
          "contrast": 9,
          "name": "neutral900",
          "value": "#b2b7b1",
        },
      ],
    },
    {
      "name": "red",
      "values": [
        {
          "contrast": 1,
          "name": "red100",
          "value": "#22110d",
        },
        {
          "contrast": 2,
          "name": "red200",
          "value": "#7d2f1e",
        },
        {
          "contrast": 3,
          "name": "red300",
          "value": "#b2361d",
        },
        {
          "contrast": 4,
          "name": "red400",
          "value": "#bf5633",
        },
        {
          "contrast": 5,
          "name": "red500",
          "value": "#c86c4e",
        },
        {
          "contrast": 6,
          "name": "red600",
          "value": "#cf7f64",
        },
        {
          "contrast": 7,
          "name": "red700",
          "value": "#d48f78",
        },
        {
          "contrast": 8,
          "name": "red800",
          "value": "#d99d89",
        },
        {
          "contrast": 9,
          "name": "red900",
          "value": "#ddab99",
        },
      ],
    },
    {
      "name": "green",
      "values": [
        {
          "contrast": 1,
          "name": "green100",
          "value": "#12160e",
        },
        {
          "contrast": 2,
          "name": "green200",
          "value": "#394e28",
        },
        {
          "contrast": 3,
          "name": "green300",
          "value": "#4b6b2f",
        },
        {
          "contrast": 4,
          "name": "green400",
          "value": "#568132",
        },
        {
          "contrast": 5,
          "name": "green500",
          "value": "#69904d",
        },
        {
          "contrast": 6,
          "name": "green600",
          "value": "#7c9d65",
        },
        {
          "contrast": 7,
          "name": "green700",
          "value": "#8ba878",
        },
        {
          "contrast": 8,
          "name": "green800",
          "value": "#99b389",
        },
        {
          "contrast": 9,
          "name": "green900",
          "value": "#a7bc9a",
        },
      ],
    },
    {
      "name": "yellow",
      "values": [
        {
          "contrast": 1,
          "name": "yellow100",
          "value": "#171610",
        },
        {
          "contrast": 2,
          "name": "yellow200",
          "value": "#4e4832",
        },
        {
          "contrast": 3,
          "name": "yellow300",
          "value": "#6b6240",
        },
        {
          "contrast": 4,
          "name": "yellow400",
          "value": "#817548",
        },
        {
          "contrast": 5,
          "name": "yellow500",
          "value": "#8f865e",
        },
        {
          "contrast": 6,
          "name": "yellow600",
          "value": "#9b9472",
        },
        {
          "contrast": 7,
          "name": "yellow700",
          "value": "#a7a083",
        },
        {
          "contrast": 8,
          "name": "yellow800",
          "value": "#b0ac93",
        },
        {
          "contrast": 9,
          "name": "yellow900",
          "value": "#bab6a1",
        },
      ],
    },
    {
      "name": "blue",
      "values": [
        {
          "contrast": 1,
          "name": "blue100",
          "value": "#131515",
        },
        {
          "contrast": 2,
          "name": "blue200",
          "value": "#3d4b4a",
        },
        {
          "contrast": 3,
          "name": "blue300",
          "value": "#516664",
        },
        {
          "contrast": 4,
          "name": "blue400",
          "value": "#578047",
        },
        {
          "contrast": 5,
          "name": "blue500",
          "value": "#6b8f5a",
        },
        {
          "contrast": 6,
          "name": "blue600",
          "value": "#7d9c6f",
        },
        {
          "contrast": 7,
          "name": "blue700",
          "value": "#8ca780",
        },
        {
          "contrast": 8,
          "name": "blue800",
          "value": "#9bb291",
        },
        {
          "contrast": 9,
          "name": "blue900",
          "value": "#a8bba0",
        },
      ],
    },
    {
      "name": "magenta",
      "values": [
        {
          "contrast": 1,
          "name": "magenta100",
          "value": "#1a1511",
        },
        {
          "contrast": 2,
          "name": "magenta200",
          "value": "#574535",
        },
        {
          "contrast": 3,
          "name": "magenta300",
          "value": "#785c46",
        },
        {
          "contrast": 4,
          "name": "magenta400",
          "value": "#817546",
        },
        {
          "contrast": 5,
          "name": "magenta500",
          "value": "#8f865d",
        },
        {
          "contrast": 6,
          "name": "magenta600",
          "value": "#9b9471",
        },
        {
          "contrast": 7,
          "name": "magenta700",
          "value": "#a6a082",
        },
        {
          "contrast": 8,
          "name": "magenta800",
          "value": "#b0ac92",
        },
        {
          "contrast": 9,
          "name": "magenta900",
          "value": "#bab7a1",
        },
      ],
    },
    {
      "name": "cyan",
      "values": [
        {
          "contrast": 1,
          "name": "cyan100",
          "value": "#131517",
        },
        {
          "contrast": 2,
          "name": "cyan200",
          "value": "#3d4a51",
        },
        {
          "contrast": 3,
          "name": "cyan300",
          "value": "#506570",
        },
        {
          "contrast": 4,
          "name": "cyan400",
          "value": "#617e3d",
        },
        {
          "contrast": 5,
          "name": "cyan500",
          "value": "#738e56",
        },
        {
          "contrast": 6,
          "name": "cyan600",
          "value": "#849b6c",
        },
        {
          "contrast": 7,
          "name": "cyan700",
          "value": "#92a77e",
        },
        {
          "contrast": 8,
          "name": "cyan800",
          "value": "#9fb18f",
        },
        {
          "contrast": 9,
          "name": "cyan900",
          "value": "#abba9d",
        },
      ],
    },
    {
      "name": "orange",
      "values": [
        {
          "contrast": 1,
          "name": "orange100",
          "value": "#1a150d",
        },
        {
          "contrast": 2,
          "name": "orange200",
          "value": "#574624",
        },
        {
          "contrast": 3,
          "name": "orange300",
          "value": "#795e2a",
        },
        {
          "contrast": 4,
          "name": "orange400",
          "value": "#917032",
        },
        {
          "contrast": 5,
          "name": "orange500",
          "value": "#9e814c",
        },
        {
          "contrast": 6,
          "name": "orange600",
          "value": "#a89063",
        },
        {
          "contrast": 7,
          "name": "orange700",
          "value": "#b29d76",
        },
        {
          "contrast": 8,
          "name": "orange800",
          "value": "#baa988",
        },
        {
          "contrast": 9,
          "name": "orange900",
          "value": "#c2b498",
        },
      ],
    },
  ],
  "terminal": [
    "#11160e",
    "#7d2f1e",
    "#394e28",
    "#4e4832",
    "#3d4b4a",
    "#574535",
    "#3d4a51",
    "#57655e",
    "#3f4b42",
    "#cf7f64",
    "#7c9d65",
    "#9b9472",
    "#7d9c6f",
    "#9b9471",
    "#849b6c",
    "#8f958c",
  ],
}
`;

exports[`generateColorScheme() > returns the correct values 2`] = `
{
  "contrastColors": [
    {
      "background": "#15150d",
    },
    {
      "name": "neutral",
      "values": [
        {
          "contrast": 1,
          "name": "neutral100",
          "value": "#16160d",
        },
        {
          "contrast": 2,
          "name": "neutral200",
          "value": "#424a3e",
        },
        {
          "contrast": 3,
          "name": "neutral300",
          "value": "#586557",
        },
        {
          "contrast": 4,
          "name": "neutral400",
          "value": "#747768",
        },
        {
          "contrast": 5,
          "name": "neutral500",
          "value": "#84877a",
        },
        {
          "contrast": 6,
          "name": "neutral600",
          "value": "#919589",
        },
        {
          "contrast": 7,
          "name": "neutral700",
          "value": "#9ea198",
        },
        {
          "contrast": 8,
          "name": "neutral800",
          "value": "#aaaca4",
        },
        {
          "contrast": 9,
          "name": "neutral900",
          "value": "#b4b7af",
        },
      ],
    },
    {
      "name": "red",
      "values": [
        {
          "contrast": 1,
          "name": "red100",
          "value": "#221111",
        },
        {
          "contrast": 2,
          "name": "red200",
          "value": "#7c2e32",
        },
        {
          "contrast": 3,
          "name": "red300",
          "value": "#b3323e",
        },
        {
          "contrast": 4,
          "name": "red400",
          "value": "#c64d4d",
        },
        {
          "contrast": 5,
          "name": "red500",
          "value": "#ce6664",
        },
        {
          "contrast": 6,
          "name": "red600",
          "value": "#d47a77",
        },
        {
          "contrast": 7,
          "name": "red700",
          "value": "#d98b88",
        },
        {
          "contrast": 8,
          "name": "red800",
          "value": "#dd9a97",
        },
        {
          "contrast": 9,
          "name": "red900",
          "value": "#e0a8a5",
        },
      ],
    },
    {
      "name": "green",
      "values": [
        {
          "contrast": 1,
          "name": "green100",
          "value": "#151611",
        },
        {
          "contrast": 2,
          "name": "green200",
          "value": "#454b35",
        },
        {
          "contrast": 3,
          "name": "green300",
          "value": "#5e6545",
        },
        {
          "contrast": 4,
          "name": "green400",
          "value": "#717959",
        },
        {
          "contrast": 5,
          "name": "green500",
          "value": "#81896d",
        },
        {
          "contrast": 6,
          "name": "green600",
          "value": "#90967f",
        },
        {
          "contrast": 7,
          "name": "green700",
          "value": "#9da28e",
        },
        {
          "contrast": 8,
          "name": "green800",
          "value": "#a8ad9c",
        },
        {
          "contrast": 9,
          "name": "green900",
          "value": "#b3b7a9",
        },
      ],
    },
    {
      "name": "yellow",
      "values": [
        {
          "contrast": 1,
          "name": "yellow100",
          "value": "#161510",
        },
        {
          "contrast": 2,
          "name": "yellow200",
          "value": "#4a4a33",
        },
        {
          "contrast": 3,
          "name": "yellow300",
          "value": "#646442",
        },
        {
          "contrast": 4,
          "name": "yellow400",
          "value": "#797663",
        },
        {
          "contrast": 5,
          "name": "yellow500",
          "value": "#898676",
        },
        {
          "contrast": 6,
          "name": "yellow600",
          "value": "#969486",
        },
        {
          "contrast": 7,
          "name": "yellow700",
          "value": "#a2a095",
        },
        {
          "contrast": 8,
          "name": "yellow800",
          "value": "#adaba2",
        },
        {
          "contrast": 9,
          "name": "yellow900",
          "value": "#b7b6ae",
        },
      ],
    },
    {
      "name": "blue",
      "values": [
        {
          "contrast": 1,
          "name": "blue100",
          "value": "#151516",
        },
        {
          "contrast": 2,
          "name": "blue200",
          "value": "#46484a",
        },
        {
          "contrast": 3,
          "name": "blue300",
          "value": "#5e6365",
        },
        {
          "contrast": 4,
          "name": "blue400",
          "value": "#887254",
        },
        {
          "contrast": 5,
          "name": "blue500",
          "value": "#958369",
        },
        {
          "contrast": 6,
          "name": "blue600",
          "value": "#a1917c",
        },
        {
          "contrast": 7,
          "name": "blue700",
          "value": "#ac9e8b",
        },
        {
          "contrast": 8,
          "name": "blue800",
          "value": "#b5aa99",
        },
        {
          "contrast": 9,
          "name": "blue900",
          "value": "#beb4a7",
        },
      ],
    },
    {
      "name": "magenta",
      "values": [
        {
          "contrast": 1,
          "name": "magenta100",
          "value": "#1c1213",
        },
        {
          "contrast": 2,
          "name": "magenta200",
          "value": "#663d40",
        },
        {
          "contrast": 3,
          "name": "magenta300",
          "value": "#8d5055",
        },
        {
          "contrast": 4,
          "name": "magenta400",
          "value": "#8a7250",
        },
        {
          "contrast": 5,
          "name": "magenta500",
          "value": "#988365",
        },
        {
          "contrast": 6,
          "name": "magenta600",
          "value": "#a39178",
        },
        {
          "contrast": 7,
          "name": "magenta700",
          "value": "#ad9e88",
        },
        {
          "contrast": 8,
          "name": "magenta800",
          "value": "#b6a997",
        },
        {
          "contrast": 9,
          "name": "magenta900",
          "value": "#bfb5a5",
        },
      ],
    },
    {
      "name": "cyan",
      "values": [
        {
          "contrast": 1,
          "name": "cyan100",
          "value": "#141615",
        },
        {
          "contrast": 2,
          "name": "cyan200",
          "value": "#414a49",
        },
        {
          "contrast": 3,
          "name": "cyan300",
          "value": "#576564",
        },
        {
          "contrast": 4,
          "name": "cyan400",
          "value": "#647a71",
        },
        {
          "contrast": 5,
          "name": "cyan500",
          "value": "#768a82",
        },
        {
          "contrast": 6,
          "name": "cyan600",
          "value": "#879791",
        },
        {
          "contrast": 7,
          "name": "cyan700",
          "value": "#95a39e",
        },
        {
          "contrast": 8,
          "name": "cyan800",
          "value": "#a2aea9",
        },
        {
          "contrast": 9,
          "name": "cyan900",
          "value": "#aeb8b4",
        },
      ],
    },
    {
      "name": "orange",
      "values": [
        {
          "contrast": 1,
          "name": "orange100",
          "value": "#1c130e",
        },
        {
          "contrast": 2,
          "name": "orange200",
          "value": "#653f29",
        },
        {
          "contrast": 3,
          "name": "orange300",
          "value": "#8f5331",
        },
        {
          "contrast": 4,
          "name": "orange400",
          "value": "#a36746",
        },
        {
          "contrast": 5,
          "name": "orange500",
          "value": "#ae7a5d",
        },
        {
          "contrast": 6,
          "name": "orange600",
          "value": "#b68a71",
        },
        {
          "contrast": 7,
          "name": "orange700",
          "value": "#bf9882",
        },
        {
          "contrast": 8,
          "name": "orange800",
          "value": "#c6a492",
        },
        {
          "contrast": 9,
          "name": "orange900",
          "value": "#ccb1a1",
        },
      ],
    },
  ],
  "terminal": [
    "#15150d",
    "#7c2e32",
    "#454b35",
    "#4a4a33",
    "#46484a",
    "#663d40",
    "#414a49",
    "#586557",
    "#424a3e",
    "#d47a77",
    "#90967f",
    "#969486",
    "#a1917c",
    "#a39178",
    "#879791",
    "#919589",
  ],
}
`;
